model/src/main/kotlin/Repository.kt

```diff
@@ -48,4 +48,17 @@ data class Repository(
          * The configuration of the repository, parsed from the ".ort.yml" file.
          */
         val config: RepositoryConfiguration
-)
+) {
+    /**
+     * Return the path of [vcs] relative to [Repository.vcs], or null if [vcs] is neither [Repository.vcs] nor contained
+     * in [nestedRepositories].
+     */
+    fun getVcsPath(vcs: VcsInfo): String? {
+        fun VcsInfo.similar(other: VcsInfo) =
+                type.equals(other.type, true) && url == other.url && revision == other.revision
```

Comment:
I wonder whether we really need to compare case-insensitively here as the type is always set by our own / the same code.

Reply:
Done, to be on the safe side I normalize the input vcs now.